cmd/helmhook/hook/addon.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright (C) 2022-2024 ApeCloud Co., Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hook
+
+import (
+	"context"
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	errors "sigs.k8s.io/controller-runtime/pkg/client"
+
+	extensionsv1alpha1 "github.com/apecloud/kubeblocks/apis/extensions/v1alpha1"
+	"github.com/apecloud/kubeblocks/pkg/client/clientset/versioned"
+)
+
+type Addon struct {
+	BasedHandler
+
+	AddonEnabledUpdated bool
+}
+
+const (
+	helmResourcePolicyKey  = "helm.sh/resource-policy"
+	helmResourcePolicyKeep = "keep"
+)
+
+func (p *Addon) IsSkip(*UpgradeContext) (bool, error) {
+	return p.AddonEnabledUpdated, nil
```

Comment:
If addEnabled is false,
then p.AddonEnabledUpdated is false,
then IsSkip() return false,
then Handle() will be executed.

Is above logic expected?

Reply:
The processing logic here is not easy to understand. The handle of the addon is prohibited addon updates.

Reply:
Should rename addonEnabled & p.AddonEnabledUpdated, since it's confusing

Reply:
rename addonEnabled to keepAddons